src/Features/CSharp/Portable/CodeCleanup/CSharpCodeCleanupService.cs

```diff
@@ -0,0 +1,190 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeCleanup;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.OrganizeImports;
+using Microsoft.CodeAnalysis.RemoveUnnecessaryImports;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Shared.Utilities;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.CSharp.CodeCleanup
+{
+    [ExportLanguageService(typeof(ICodeCleanupService), LanguageNames.CSharp), Shared]
+    internal class CSharpCodeCleanupService : ICodeCleanupService
+    {
+        /// <summary>
+        /// Maps format document code cleanup options to DiagnosticId[]
+        /// </summary>
+        private static ImmutableDictionary<PerLanguageOption<bool>, ImmutableArray<string>> _optionDiagnosticsMappings = GetCodeCleanupOptionMapping();
+
+        private readonly ICodeFixService _codeFixService;
+
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public CSharpCodeCleanupService(
+            // will remove the AllowDefault once CodeFixService is moved to Features
+            // https://github.com/dotnet/roslyn/issues/27369
+            [Import(AllowDefault = true)] ICodeFixService codeFixService)
+        {
+            _codeFixService = codeFixService;
+        }
+
+        private static ImmutableDictionary<PerLanguageOption<bool>, ImmutableArray<string>> GetCodeCleanupOptionMapping()
+        {
+            var dictionary = new Dictionary<PerLanguageOption<bool>, ImmutableArray<string>>()
```

Comment:
you can use ImmutableDictionaryBuilder rather than use Dictionary and do ToImmutableDictionary? is it to use built in dictionary object intializer?

Reply:
yeah, i don't find a way to use dictionary initializer for Immutable dictionary